### PR TITLE
chore: upgrade @nestjs v8

### LINKED ISSUE #272: Support NestJS v8
Please, add NestJS V8 support.

### PR DESCRIPTION
**Pull request by 9renpoto**
closes #272

### CODE REVIEW

File: example/__tests__/app.e2e-spec.ts [hunk 1]
```diff
@@ -18,10 +18,7 @@ describe('AppModule', () => {
     app = moduleFixture.createNestApplication();
     await app.init();
 
-    const module: GraphQLModule =
-      moduleFixture.get<GraphQLModule>(GraphQLModule);
-    // apolloServer is protected, we need to cast module to any to get it
-    apolloClient = createTestClient((module as any).apolloServer);
+    apolloClient = getApolloServer(moduleFixture);
```
Comment (9renpoto) [inline example/__tests__/app.e2e-spec.ts:21]: refs: https://github.com/nestjs/graphql/blob/8.0.2/tests/e2e/code-first.spec.ts#L19

File: example/__tests__/app.e2e-spec.ts [hunk 2]
```diff
@@ -1,14 +1,14 @@
 import { Test, TestingModule } from '@nestjs/testing';
 import * as request from 'supertest';
 import { INestApplication } from '@nestjs/common';
-import { GraphQLModule } from '@nestjs/graphql';
-import { createTestClient } from 'apollo-server-testing';
```
Comment (9renpoto) [inline example/__tests__/app.e2e-spec.ts:5]: > This package has been deprecated

https://www.npmjs.com/package/apollo-server-testing

**Comment by 9renpoto** [comment]
@mentos1386 hi, can you review ? :)

File: test/websockets/class.e2e.spec.ts [hunk 3]
```diff
@@ -29,15 +28,10 @@ describe('Websockets:Class', () => {
...
-    socket = io.connect('http://localhost:4455');
-  });
-
-  afterEach(() => {
-    socket.disconnect();
   });
 
   it(`emit:test_error`, async () => {
-    const socket = io.connect('http://localhost:4444');
+    const socket = io('http://localhost:4444');
```
Comment (9renpoto) [inline test/websockets/class.e2e.spec.ts:34]: https://socket.io/docs/v4/client-initialization/

File: package.json [hunk 4]
```diff
@@ -36,40 +36,42 @@
...
+    "@nestjs/platform-express": "8.0.6",
+    "@nestjs/platform-socket.io": "8.0.6",
+    "@nestjs/testing": "8.0.6",
+    "@nestjs/websockets": "8.0.6",
+    "@types/jest": "27.0.1",
+    "@types/node": "16.6.1",
     "@types/supertest": "2.0.11",
     "apollo-server-express": "2.25.2",
-    "apollo-server-testing": "2.25.2",
+    "class-transformer": "^0.4.0",
```
Comment (mentos1386) [inline package.json:60]: I don't see this new dependencies used anywhere. Are they needed?

**Review by mentos1386 — APPROVED** [review]
Other then dependencies comment, LGTM :) Thanks for taking your time contributing to this project 💚

File: package.json (hunk 4 above)
Comment (9renpoto) [reply package.json:60]:
```
npm WARN @nestjs/mapped-types@0.0.5 requires a peer of class-transformer@^0.2.3 but none is installed. You must install peer dependencies yourself.
npm WARN @nestjs/mapped-types@0.0.5 requires a peer of class-validator@^0.11.1 || ^0.12.0 but none is installed. You must install peer dependencies yourself.
```

**Pull request merged**